File: algo/reverse-string-ii/reverseStr.go
func reverse(bs []byte, start, end int) {
    if end >= len(bs) {
        end = len(bs) - 1
    }
    
    for start < end {
        bs[start], bs[end] = bs[end], bs[start]
        start++
        end--
    }
}

func reverseStr(s string, k int) string {
    start := 0
    retStr := []byte(s)
    
    for start < len(s) { 
        reverse(retStr, start, start+k-1)
        start += 2 * k
    }
    
    return string(retStr)
}

File: algo/reverse-string-ii/reverseStr.py
class Solution(object):
    def reverseStr(self, s, k):
        """
        :type s: str
        :type k: int
        :rtype: str
        """
        retStr = ''
        start = 0
        
        while start < len(s):
            retStr += s[start:start+k][::-1] + s[start+k:start+2*k]
            start += 2*k
            
        return retStr
